src/query/somepath.go

```diff
@@ -4,60 +4,88 @@ import "fmt"
 
 import "github.com/thought-machine/please/src/core"
 
-// SomePath finds and returns a path between two targets.
+// SomePath finds and returns a path between two targets, or between one and a set of targets.
 // Useful for a "why on earth do I depend on this thing" type query.
-func SomePath(graph *core.BuildGraph, label1 core.BuildLabel, label2 core.BuildLabel) {
-	// Awkwardly either target can be :all. This is an extremely useful idiom though so despite
-	// trickiness is worth supporting.
-	// Of course this calculation is also quadratic but it's not very obvious how to avoid that.
-	if label1.IsAllTargets() {
-		for _, target := range graph.PackageOrDie(label1).AllTargets() {
-			if querySomePath1(graph, target, label2, false) {
+func SomePath(graph *core.BuildGraph, from, to []core.BuildLabel) {
+	from = expandAllTargets(graph, from)
+	to = expandAllTargets(graph, to)
+	s := somepath{
+		memo: map[core.BuildLabel]map[core.BuildLabel]struct{}{},
+	}
+	for _, l1 := range expandAllTargets(graph, from) {
```

Comment:
Are we re-expanding from?

Reply:
it's always best to call functions again just to be sure they've really done what they're meant to

Reply:
I sense some trust issues. You should really work on that.